app/services/service_listeners/publishing_api_pusher.rb

```diff
@@ -31,6 +31,11 @@ def push(event:, options: {})
       end
 
       handle_html_attachments(event)
+      if edition.try(:document).try(:content_id) &&
```

Comment:
There's a lot of `try`(ing) going on here. If the `Edition` doesn't have a `Document` and/or the `Document` doesn't have a `content_id` then that is an exceptional state and we should `raise` rather than ignore it by trying. With that removed we could then refactor this out into a few smaller methods that could be named to express the logic more clearly.